crates/router/src/core/admin.rs

```diff
@@ -212,6 +266,19 @@ pub async fn merchant_account_update(
             .attach_printable("Invalid routing algorithm given")?;
     }
 
+    let encrypt_string = |inner: Option<Secret<String>>| async {
+        inner
+            .async_map(|value| domain_types::encrypt(value, &key))
+            .await
+            .transpose()
+    };
+
+    let encrypt_value = |inner: OptionalSecretValue| async {
+        inner
+            .async_map(|value| domain_types::encrypt(value, &key))
+            .await
+            .transpose()
+    };
```

Comment:
This seems to be repeated. can be moved to some utils

Reply:
This cannot be moved into a function of its own. the reason being
- The key is being moved from the current context into the closure
- The the only solution that is left is: we create a function that looks something like this:
```rust
fn encrypt_closure_maker<T: Clone, S: Strategy<T>>(key: Vec<u8>) -> 
	(impl Fn(Option<Secret<T, S>>)) -> 
		(impl futures::Future<Output = 
			CustomResult<Option<Encryptable<Secret<T, S>>>, errors::CryptoError>>
		)
	)
where
	Secret<T, S>: Send
```
These closure that this function will yield cannot be generic as they themselves had uncertain fields or returns. this only possible way is to do `impl ...`. But we cannot have and `impl` inside an `impl` statement. Thus, for now we can only work with the manual closures that we are creating here.